### ISSUE TITLE
output tables take advantage of payroll taxes

### ISSUE THREAD

**Issue by MattHJensen** [OpenSourcePolicyCenter/webapp-public#62]
We can now calculate payroll taxes. For our tables to take advantage of this we should make some modifications:

Revenue table:
- [x] header -> Total Liabilities Change by Calendar Year
- [x] the total revenue row should be relabelled "Individual Income Tax Liability Change"
- [x] a new row should be labelled "Payroll Tax Liability Change" 
- [x] a new row should be labelled "Combined Payroll and Individual Income Tax Liability Change" 

Diagnostic tables: 
- [x] the "Revenue" column should be relabelled "Individual Income Tax Liabilities"
- [x] a new column should be labelled "Payroll Tax Liabilities" 
- [x] a new column should be labelled "Combined Payroll and Individual Income Tax Liabilities

Difference tables: 
- [x] "Individual Income Tax:" should be prepended to the existing table names, so an example name would be, "INDIVIDUAL INCOME TAX: DIFFERENCE BETWEEN BASE AND USER PLANS BY AGI DECILE"
- [ ] A new set of tables should be added for "Payroll Tax:.."
- [ ] A new set of tables should be added for "Combined Payroll and Individual Income Tax:..."

**Comment by talumbau** [comment]
> a new row should be labelled "Payroll Tax Liability Change"
>  a new row should be labelled "Combined Payroll and Individual Income Tax Liability Change"

In order to add these rows, I need to know the calculation to do. For the current single row in this table, we do the calculation here:

https://github.com/OpenSourcePolicyCenter/dropQ/blob/master/dropq/dropq.py#L122

which looks like this:
```
dec_sum = (df2['tax_diff_dec']*df2['s006']).sum()
```

so just the difference in tax between the two plans for each record weighted by `s006` and then summed. What is the correct calculation for the "Payroll Tax Liability Change" row and the "Combined Payroll and Individual Income Tax Liability Change"? cc @Amy-Xu

**Comment by Amy-Xu** [comment]
@talumbau Yep I saw that line. The other two rows would be calculated in the same way but replacing `_ospctax` with `_payroll` and `_combined` respectively. It's not difficult to add those calculations in `dropq.py`, but I was hesitant because I realized in the `helper.py` this table's if-loops need to be replaced by a for loop. I don't think I want to do that without being able to test it on my computer. I was wondering do you mind make changes to `helper.py` on your end? I can just add the calculations of the two rows here in dropq.py

**Comment by talumbau** [comment]
OK, I understand what you are saying in regards to using `_payroll` and `_combined` to do the calculation for the two new rows. If you want to put the PR up on dropq that does this calculation that's fine, but if not, I will put it up soon and ask for your review. Thanks for that info!

Regarding the other part, I am not certain I understand what you are saying. Let me know if this is correct: in the webapp repo, we have a function `taxcalc_results_to_tables` in `helpers.py`. This function would have to be modified for the `fiscal_tots` table, because currently it does the following logic to set up that table for formatting:

```
        elif table_id == 'fiscal_tots':
            # todo - move these into the above TC result param constants
            row_keys = ['totals']
            row_labels = {'totals': 'Total Revenue'}
            col_labels = years
            col_formats = [ [1000000000, 'Dollars', 1] for y in years]
            table_data = {'totals': results[table_id]}
            multi_year_cells = False
```

So more would have to be added here to make sure it does the formatting properly for the two new rows. Is this what you are saying? If yes, I can do this and test it out. If I misunderstood, can you help clarify? Thanks!

**Comment by Amy-Xu** [comment]
@talumbau Sorry let me clarify it a bit more. Yes, you're definitely right on the snippet you pasted here. What I meant is, in addition to that snippet, I thought we also need to change how this fiscal_tots table is processed in this section of the same file:

```
    #FISCAL TOTS
    res.append(["#fiscal totals data"])
    ft = tax_results.get('fiscal_tots', [])
    yrs = [START_YEAR + i for i in range(0, len(ft))]
    if yrs:
        res.append(yrs)
    if ft:
        res.append(ft)
```

It seems to me this section of code processes one row instead of three rows. Maybe I'm just confusing myself. Could you take a look?

**Comment by talumbau** [comment]
> Yep I saw that line. The other two rows would be calculated in the same way but replacing _ospctax with _payroll and _combined respectively.

@Amy-Xu, in looking through your PR here:

https://github.com/open-source-economics/Tax-Calculator/pull/392/files

I'm wondering if, in the above text, you meant to say "`_fica`", instead of "`_payroll`" to indicate the column on which to do the calculations for the row called "Payroll Tax Liability Change". If you did actually mean to say "`_payroll`", then I'm having a hard time understanding what to do because I don't see that column in your latest PR. Can you clarify? Thanks!

**Comment by Amy-Xu** [comment]
@talumbau Sorry about the confusion. Yes, I meant `_fica` instead of `_payroll`.

**Comment by talumbau** [comment]
@MattHJensen following back up with this issue. In your list under "Difference Tables", it looks like you are saying that we need four more difference tables. Is that correct? Two for differences between X and Y in Payroll Tax (one for expanded percentile, one for expanded income bin), and two for differences between X and Y in the sum of ind. income tax and payroll tax. Is that correct? that seems to be the last remaining items on this issue.

**Comment by MattHJensen** [comment]
@talumbau, that is correct. Later all of these tables will get consolidated as discussed https://github.com/OpenSourcePolicyCenter/webapp-public/issues/35#issuecomment-144135847 and in the follow on comments.

**Comment by talumbau** [comment]
resolved via #94
